BikeTrack/UserService.swift

```diff
@@ -1,41 +1,44 @@
 import Foundation
 import os.log
-import Combine
 
 /**
  `UserService` provides an interface into common user account management functions.
  
  Being an observable object, `UserService` is intended to be initialised as an `ObservableObject` high up in the SwiftUI view hierarchy and observed as an EnvironmentObject by child views as required.
  Like other `[Name]Service` classes, `UserService` provides a layer of abstraction between the UI and lower level repositories, which can be swapped out at `init()`.
  */
+@MainActor
 class UserService: ObservableObject {
     private static let logger = Logger(
         subsystem: Bundle.main.bundleIdentifier ?? AppEnvironment.defaultBundleID,
         category: String(describing: UserService.self)
     )
     
     private var authRepository: AuthRepository
-    private var cancellables = Set<AnyCancellable>()
-
-    @Published private (set) var authState: AuthState = .unknown
+    private var authObserver: Task<Void, Never>?
+    
+    @Published private(set) var authState: AuthState = .unknown {
+        didSet {
+            #if DEBUG
```

Comment:
use `Self.logger.info("published authState changed to: \(authState)")` here - then there's no need for the `#if DEBUG` statement.

Reply:
Agree!  This was an artifact of doing some quick and dirty testing to make sure @Published was being triggered with the AsyncSequence.